core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -531,7 +538,32 @@ private ExternalSearcherManager createSearcherManager(SearchFactory externalSear
         try {
             try {
                 final DirectoryReader directoryReader = ElasticsearchDirectoryReader.wrap(DirectoryReader.open(indexWriter), shardId);
-                internalSearcherManager = new SearcherManager(directoryReader, new SearcherFactory());
+                internalSearcherManager = new SearcherManager(directoryReader, new SearcherFactory() {
+                    @Override
+                    public IndexSearcher newSearcher(IndexReader reader, IndexReader previousReader) throws IOException {
+                        if (breakerService != null) {
```

Comment:
I think we should specialize a new `RamAccountingSearcherFactory` that we can test in isolation to this. I think we really should make sure we unittest this.

Reply:
:+1: